enumerable_methods.rb

```diff
@@ -55,14 +59,16 @@ def my_select
   end
 
   def my_all?
-    return to_enum unless block_given?
-
     arr = self
     if arr.is_a? Array
       x = true
       i = 0
       while i < arr.length
-        x = false unless yield(arr[i])
+        if block_given?
+          x = false unless yield(arr[i])
+        else
+          x = false unless arr[i]
+        end
         i += 1
```

Comment:
**[REQUIRED]**
* When a class is passed as an argument returns true if all of the collection is a member of such class.
```
Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true if all of the collection matches the Regex.
```
Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
```
Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```